src/rebar_app_discover.erl

```diff
@@ -34,13 +34,13 @@ format_error({missing_module, Module}) ->
     io_lib:format("Module defined in app file missing: ~p~n", [Module]).
 
 merge_deps(AppInfo, State) ->
-    Profiles = rebar_state:current_profiles(State),
+    Default = rebar_state:default(State),
+    CurrentProfiles = rebar_state:current_profiles(State),
     Name = rebar_app_info:name(AppInfo),
     C = rebar_config:consult(rebar_app_info:dir(AppInfo)),
-
     AppState = rebar_state:apply_overrides(
                  rebar_state:apply_profiles(
-                   rebar_state:new(State, C, rebar_app_info:dir(AppInfo)), Profiles), Name),
+                   rebar_state:new(rebar_state:opts(State, Default), C, rebar_app_info:dir(AppInfo)), CurrentProfiles), Name),
```

Comment:
Should have a comment somewhere about why we reset defaults there. This is really non-trivial.